core/modules/views/plugins/views_plugin_style_grid.inc

```diff
@@ -12,17 +12,20 @@
  */
 class views_plugin_style_grid extends views_plugin_style {
   /**
-   * Set default options
+   * Set default options.
    */
   function option_definition() {
     $options = parent::option_definition();
-
     $options['columns'] = array('default' => '4');
```

Comment:
Yep, youre right. This was ported before we had number fields. Will need to change the validator as well.